docs/src/content/docs/book/learn-tact-in-y-minutes.mdx

```diff
@@ -0,0 +1,1484 @@
+
+## Comments
+
+```tact
+// Single-line (//) comments for occasional and casual annotations
+
+/// Documentation comments that support Markdown
+```
+
+:::note
+
+  There are also C-like multiline comments `/* ... */`, but their use is discouraged in favor of consecutive single-line documentation comments `///`.
+
+:::
+
+## "Hello, World!" {#hello-world}
+
+```tact
+// Defining a contract
+contract HelloWorld {
+    // Listens to incoming "ping" messages
+    receive("ping") {
+        // Sends a reply: "pong"
+        reply("pong");
+    }
+
+    // Listens to incoming "hello" messages
+    receive("hello") {
+        // Sends a reply: "Hello, World!"
+        reply("Hello, World!");
+    }
+}
+
+// A helper inlined function to send text messages
+inline fun reply(str: String) {
+    message(MessageParameters{
+        to: sender(),
+        value: 0,
+        mode: SendRemainingValue | SendIgnoreErrors,
+        body: str.asComment(),
+    });
+}
+```
+
+:::note
+
+  The text receivers shown above are expensive because they need to hash their contents to differentiate message bodies, and that is a costly operation.
+
+  Later, see examples of cheaper and more controlled binary receivers that expect message structs: [Functions](#functions).
+
+:::
+
+## Primitive types
+
+```tact
+fun showcase() {
+    // There are two main groups of primitive types on TON: integers and cells.
+    // All other primitive types are derivatives of those two.
+
+    // ---
+    // Integers,
+    // always 257-bit signed in runtime operations,
+    // but may have different lengths in persistent contract's state (storage)
+    // ---
+
+    let one_plus_one: Int = 1 + 1; // 2
+    let two_by_two: Int = 2 / 2;   // 1
+    let three_by_two: Int = 3 / 2; // 1, because the division operator rounds
+                                   // toward -∞, which is identical to // operator
+                                   // from Python
+
+    let one_billion = 1_000_000_000; // decimal
+    let binary_mask = 0b1_1111_1111; // binary
+    let permissions = 0o7_5_5;       // octal
+    let heHex = 0xFF80_0000_0000;    // hexadecimal
+
+    let nanoToncoin: Int = 1;    // 1 nanoToncoin = 0.000,000,001 Toncoin
+    let toncoin: Int = ton("1"); // 1 Toncoin = 1,000,000,000 nanoToncoin
+
+
+    // ---
+    // Booleans: true and false.
+    // They take only 1 bit in persistent storage.
+    // ---
+
+    let factual: Bool = !!(true || false);
+    let fax: Bool = true && factual;
+
+
+    // ---
+    // Addresses of smart contracts,
+    // deterministically obtained by combining the initial code and initial data.
+    // ---
+
+    // This is cheap:
+    let myAddr: Address = myAddress();
+
+    // This is very expensive on-chain, do such calculations off-chain:
+    let myAddr2: Address = contractAddress(initOf MyContract());
```

Comment:
Let's remove this part. It will be consufing for newcomers. It needs to be moved to the Address type page. Arguably, you can add it to the gas optimizations page as an ant-pattern